gnovm/pkg/doc/print.go

```diff
@@ -159,195 +141,70 @@ func (pkg *pkgPrinter) oneLineNodeDepth(node ast.Node, depth int) string {
 	case nil:
 		return ""
 
-	case *ast.GenDecl:
+	case *JSONValueDecl:
 		// Formats const and var declarations.
 		trailer := ""
-		if len(n.Specs) > 1 {
+		if len(n.Values) > 1 {
 			trailer = " " + dotDotDot
 		}
 
 		// Find the first relevant spec.
-		typ := ""
-		for i, spec := range n.Specs {
-			valueSpec := spec.(*ast.ValueSpec) // Must succeed; we can't mix types in one GenDecl.
-
+		typ := n.Values[0].Type
+		for _, spec := range n.Values {
 			// The type name may carry over from a previous specification in the
 			// case of constants and iota.
-			if valueSpec.Type != nil {
-				typ = fmt.Sprintf(" %s", pkg.oneLineNodeDepth(valueSpec.Type, depth))
-			} else if len(valueSpec.Values) > 0 {
-				typ = ""
+			if spec.Type != "" {
+				typ = spec.Type
 			}
 
-			if !pkg.isExported(valueSpec.Names[0].Name) {
+			if !pkg.isExported(spec.Name) {
 				continue
 			}
-			val := ""
-			if i < len(valueSpec.Values) && valueSpec.Values[i] != nil {
-				val = fmt.Sprintf(" = %s", pkg.oneLineNodeDepth(valueSpec.Values[i], depth))
+			var token string
+			if n.Const {
+				token = "const"
+			} else {
+				token = "var"
 			}
-			return fmt.Sprintf("%s %s%s%s%s", n.Tok, valueSpec.Names[0], typ, val, trailer)
+			return fmt.Sprintf("%s %s %s%s", token, spec.Name, typ, trailer)
 		}
 		return ""
 
-	case *ast.FuncDecl:
-		// Formats func declarations.
-		name := n.Name.Name
-		recv := pkg.oneLineNodeDepth(n.Recv, depth)
-		if len(recv) > 0 {
-			recv = "(" + recv + ") "
-		}
-		fnc := pkg.oneLineNodeDepth(n.Type, depth)
-		fnc = strings.TrimPrefix(fnc, "func")
-		return fmt.Sprintf("func %s%s%s", recv, name, fnc)
-
-	case *ast.TypeSpec:
+	case *JSONType:
```

Comment:
I guess this doesn't handle recursion, like `type A map[string]struct { A int }`. :thinking: 

Maybe we can have the oneline representation be an option for JSON generation?

Reply:
> Maybe we can have the oneline representation be an option for JSON generation?

Do you mean an option so that when the `JSONDocumentation` is generated, that the [`Type`](https://github.com/gnolang/gno/blob/1ffb7adb3ecc1847dbbb1952b723466a49333deb/gnovm/pkg/doc/json_doc.go#L73) in the `JSONType` is truncated like `map[string]struct { ... }` ?

Reply:
I suppose that we should use the `Short` flag, but from what I can see it is [always false](https://github.com/gnolang/gno/blob/ced84bf72debec5250d7f6fc257d7bbeb726f569/gnovm/cmd/gno/doc.go#L122).